hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmUtil.java

```diff
@@ -377,33 +362,48 @@ else if ( domainParamBinding.getBindType() instanceof BasicValuedMapping ) {
 					final BasicValueConverter valueConverter = jdbcMapping == null ? null : jdbcMapping.getValueConverter();
 					if ( valueConverter != null ) {
 						final Object convertedValue = valueConverter.toRelationalValue( domainParamBinding.getBindValue() );
+						if ( convertedValue != null ) {
```

Comment:
Maybe I'm missing something, but wouldn't this become a problem if e.g. `"val1"` is converted to `null` i.e. `domainParamBinding.getBindValue()` is `"val1"` and `convertedValue` is `null`, leading to the creation of a `JdbcParameterBindingImpl` with the unconverted value?

Reply:
Now we'd miss a `JdbcParameterBindingImpl` for values that get converted to `null`.

Reply:
:( going to fix this.